lib/stanford/importer/purl_parser.rb

```diff
@@ -0,0 +1,175 @@
+module Stanford
+  module Importer
+    # rubocop:disable Metrics/ClassLength
+    class PurlParser
+      DC_NS = { 'dc'.freeze => 'http://purl.org/dc/elements/1.1/'.freeze }.freeze
+
+      attr_reader :xml
+
+      def initialize(xml_file)
+        @xml = Nokogiri::XML(xml_file)
+      end
+
+      def oai
+        @oai ||= xml.xpath('//oai_dc:dc', 'oai_dc' => "http://www.openarchives.org/OAI/2.0/oai_dc/")
+      end
+
+      def model
+        @model ||= if collection?
+                     Collection
+                   elsif image?
+                     'Image'
+                   else
+                     'ETD'
+                   end
+      end
+
+      def origin_text
+        'Imported from PURL to local RDF profile by Hybox'.freeze
+      end
+
+      def collection?
+        xml.xpath('//identityMetadata/objectType').any? { |c| c.text == 'collection' }
+      end
+
+      # For now the only things we import are collections and
+      # images, so if it's not a collection, assume it's an image.
+      def image?
```

Comment:
Sorry, what I meant was that this logic prevents getting to the `else` clause of `#model`.